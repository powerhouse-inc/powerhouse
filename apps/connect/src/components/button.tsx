const Button: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = ({
    className,
    ...props
}) => {
    return (
        <button
            className={`rounded-md bg-accent-5/20 px-4 py-[6px] text-accent-5 shadow-button hover:bg-accent-3 hover:shadow-none ${className}`}
            {...props}
        />
    );
};

export default Button;
