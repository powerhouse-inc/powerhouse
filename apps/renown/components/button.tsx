export interface ButtonProps
    extends React.DetailedHTMLProps<
            React.ButtonHTMLAttributes<HTMLButtonElement>,
            HTMLButtonElement
        >,
        React.AriaAttributes {
    primary?: boolean;
    secondary?: boolean;
}

const Button: React.FC<ButtonProps> = ({
    className,
    primary,
    secondary,
    ...props
}) => {
    return (
        <button
            className={`${
                primary
                    ? "bg-action text-white hover:bg-action/75"
                    : secondary
                    ? "bg-neutral-2 text-neutral-4 hover:bg-neutral-4/10"
                    : ""
            } 
            font-semibold disabled:opacity-60 disabled:pointer-events-none
            transition-colors rounded-md p-2 ${className}`}
            {...props}
        />
    );
};

export default Button;
