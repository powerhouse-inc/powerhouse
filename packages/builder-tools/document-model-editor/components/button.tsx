import type { ComponentPropsWithRef } from "react";
import { forwardRef } from "react";
import { cn } from "../utils/style.js";

type Props = ComponentPropsWithRef<"button">;

export const Button = forwardRef<HTMLButtonElement, Props>((props, ref) => {
  const { className, ...rest } = props;
  return (
    <button
      ref={ref}
      {...rest}
      className={cn(
        "h-10 whitespace-nowrap rounded-md border border-gray-200 bg-gray-50 px-4 py-2 text-sm font-medium text-gray-800 transition-colors hover:bg-gray-100 hover:text-gray-900 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
        className,
      )}
    />
  );
});
