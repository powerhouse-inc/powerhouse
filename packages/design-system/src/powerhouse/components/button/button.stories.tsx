import { Meta, StoryObj } from "@storybook/react";
import { Button } from ".";
import { Icon } from "..";

const meta = {
  title: "Powerhouse/Components/Button",
  component: Button,
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  args: {
    children: "Button",
  },
  decorators: [
    (Story) => (
      <div className="grid h-screen place-items-center bg-white">
        <Story />
      </div>
    ),
  ],
};

export const Default: Story = {
  ...Template,
};

export const DefaultHover: Story = {
  ...Default,
  parameters: {
    pseudo: {
      hover: true,
    },
  },
};

export const DefaultActive: Story = {
  ...Default,
  parameters: {
    pseudo: {
      active: true,
    },
  },
};

export const DefaultDisabled: Story = {
  ...Default,
  args: {
    ...Default.args,
    disabled: true,
  },
};

export const Light: Story = {
  ...Template,
  args: {
    ...Template.args,
    color: "light",
  },
};

export const LightHover: Story = {
  ...Light,
  parameters: {
    pseudo: {
      hover: true,
    },
  },
};

export const LightActive: Story = {
  ...Light,
  parameters: {
    pseudo: {
      active: true,
    },
  },
};

export const LightDisabled: Story = {
  ...Light,
  args: {
    ...Light.args,
    disabled: true,
  },
};

export const Red: Story = {
  ...Template,
  args: {
    ...Template.args,
    color: "red",
  },
};

export const RedHover: Story = {
  ...Red,
  parameters: {
    pseudo: {
      hover: true,
    },
  },
};

export const RedActive: Story = {
  ...Red,
  parameters: {
    pseudo: {
      active: true,
    },
  },
};

export const RedDisabled: Story = {
  ...Red,
  args: {
    ...Red.args,
    disabled: true,
  },
};

export const Blue: Story = {
  ...Template,
  args: {
    ...Template.args,
    color: "blue",
  },
};

export const BlueHover: Story = {
  ...Blue,
  parameters: {
    pseudo: {
      hover: true,
    },
  },
};

export const BlueActive: Story = {
  ...Blue,
  parameters: {
    pseudo: {
      active: true,
    },
  },
};

export const BlueDisabled: Story = {
  ...Blue,
  args: {
    ...Blue.args,
    disabled: true,
  },
};

export const Small: Story = {
  ...Template,
  args: {
    ...Template.args,
    size: "small",
  },
};

export const WithIcon: Story = {
  ...Template,
  args: {
    ...Template.args,
    icon: <Icon name="Plus" size={20} />,
  },
};

export const SmallWithIcon: Story = {
  ...Small,
  args: {
    ...Small.args,
    icon: <Icon name="Plus" size={16} />,
  },
};
