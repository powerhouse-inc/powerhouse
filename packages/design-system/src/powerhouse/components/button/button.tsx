import { ComponentPropsWithRef, ForwardedRef, forwardRef } from "react";
import { twJoin, twMerge } from "tailwind-merge";

export type ButtonProps = ComponentPropsWithRef<"button"> & {
  readonly color?: "light" | "dark" | "red" | "blue";
  readonly size?: "small" | "medium";
  readonly icon?: React.JSX.Element;
  readonly iconPosition?: "left" | "right";
};

export const Button = forwardRef(function Button(
  props: ButtonProps,
  ref: ForwardedRef<HTMLButtonElement>,
) {
  const {
    color = "dark",
    size = "medium",
    className = "",
    children,
    icon,
    iconPosition = "left",
    ...delegatedProps
  } = props;

  const sizeStyles = {
    small: "px-2 py-1.5 text-xs rounded-md font-medium",
    medium: "px-6 py-3 text-base rounded-xl font-semibold tracking-wide",
  };

  const colorStyles = {
    light:
      "bg-gray-200 text-gray-600 hover:text-gray-700 hover:border-gray-300 active:border-slate-100 active:text-gray-600 disabled:text-gray-400",
    dark: "bg-gray-800 text-slate-50 hover:bg-slate-800 active:border-slate-700 disabled:bg-gray-300 disabled:text-slate-100",
    red: "bg-red-900 text-slate-50 hover:opacity-80 active:border-red-800 disabled:text-red-400 disabled:opacity-100",
    blue: "bg-blue-900 text-slate-50 hover:opacity-80 active:border-blue-800 disabled:text-blue-400 disabled:opacity-100",
  };

  const colorAndSizeStyle = twJoin(colorStyles[color], sizeStyles[size]);

  const finalClassName = twMerge(
    "flex items-center justify-center gap-2 border border-none outline-none transition disabled:cursor-not-allowed",
    colorAndSizeStyle,
    className,
  );

  return (
    <button className={finalClassName} ref={ref} {...delegatedProps}>
      {iconPosition === "left" && icon}
      {children}
      {iconPosition === "right" && icon}
    </button>
  );
});
