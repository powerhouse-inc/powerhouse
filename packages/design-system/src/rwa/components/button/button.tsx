import { Button, ButtonProps, mergeClassNameProps } from "@/powerhouse";
import React from "react";

export const RWAButton: React.FC<ButtonProps> = (props) => {
  return (
    <Button
      color="light"
      size="small"
      {...mergeClassNameProps(
        props,
        "border border-solid border-gray-300 bg-white px-3 py-1 text-sm text-gray-900",
      )}
    >
      {props.children}
    </Button>
  );
};
