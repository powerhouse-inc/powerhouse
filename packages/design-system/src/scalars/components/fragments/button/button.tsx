import { cn } from "#scalars";
import { Slot } from "@radix-ui/react-slot";
import { cva, type VariantProps } from "class-variance-authority";
import React from "react";

const buttonVariants = cva(
  cn(
    "inline-flex items-center justify-center gap-2",
    "whitespace-nowrap rounded-md text-sm font-medium",
    "transition-colors",
    "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2",
    "disabled:pointer-events-none",
    "[&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0",
  ),
  {
    variants: {
      variant: {
        default: "",
        outline:
          "border border-input bg-background hover:bg-accent hover:text-accent-foreground",
        ghost: "hover:bg-accent hover:text-accent-foreground",
      },
    },
    defaultVariants: {
      variant: "default",
    },
  },
);

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  asChild?: boolean;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ variant, className, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return (
      <Comp
        className={cn(buttonVariants({ variant, className }))}
        ref={ref}
        {...props}
      />
    );
  },
);

export { Button, buttonVariants };
