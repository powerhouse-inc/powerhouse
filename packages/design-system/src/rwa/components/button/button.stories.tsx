import { Icon } from "@/powerhouse";
import type { Meta, StoryObj } from "@storybook/react";
import { RWAButton } from "./button";

const meta: Meta<typeof RWAButton> = {
  title: "RWA/Components/RWAButton",
  component: RWAButton,
  argTypes: {
    children: { control: "text" },
    icon: { control: "object" },
    iconPosition: {
      options: ["left", "right"],
      control: { type: "radio" },
    },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    children: "Edit Transaction",
    iconPosition: "right",
    icon: <Icon name="Pencil" size={16} />,
  },
};
