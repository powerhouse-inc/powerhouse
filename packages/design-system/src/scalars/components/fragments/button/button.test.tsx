import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { Button } from "./button.js";

describe("Button Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
