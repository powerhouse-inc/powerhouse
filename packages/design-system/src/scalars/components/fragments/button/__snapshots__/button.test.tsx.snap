// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
  />
</DocumentFragment>
`;
